src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -4360,30 +5033,539 @@ public static void ToLowerInvariant(string s, string expected)
+            Assert.Equal("", emptyString.ToLowerInvariant());
+
+            ReadOnlySpan<char> source = emptyString.AsSpan();                       
+            Assert.Equal(source.Length, source.ToLower(destination, CultureInfo.CurrentCulture));
+            Assert.Equal(source.Length, source.ToLowerInvariant(destination));
+            Assert.Equal(expectedDestination, destination.ToArray());
+            Assert.Equal(expectedSource, a);
+
+            emptyString = string.Empty;            
+            Assert.Equal("", emptyString.ToLowerInvariant()); 
+         
+            source = emptyString.AsSpan();
+            Assert.Equal(source.Length, source.ToLower(destination, CultureInfo.CurrentCulture));
+            Assert.Equal(source.Length, source.ToLowerInvariant(destination));
+            Assert.Equal(expectedDestination, destination.ToArray());
+            Assert.Equal(expectedSource, a);
         }
 
-        [Theory]
-        [InlineData("hello", "HELLO")]
-        [InlineData("HELLO", "HELLO")]
-        [InlineData("hElLo", "HELLO")]
-        [InlineData("HeLlO", "HELLO")]
-        [InlineData("", "")]
-        public static void ToUpper(string s, string expected)
+        [Fact]
+        public static void SameSpanToLower()
         {
-            Assert.Equal(expected, s.ToUpper());
-            
-            Span<char> destination = new char[s.Length];
-            Assert.Equal(s.Length, s.AsSpan().ToUpper(destination, CultureInfo.CurrentCulture));
+            var expected = new char[3] { 'a', 'b', 'c' };
+            var a = new char[3] { 'a', 'B', 'c' };
+            string s1 = new string(a);
+            Assert.Equal(expected, s1.ToLower(CultureInfo.CurrentCulture).ToArray());
+            Assert.Equal(expected, s1.ToLowerInvariant().ToArray());
+            {
+                ReadOnlySpan<char> source = a;
+                Span<char> destination = a;
+                Assert.Equal(source.Length, source.ToLower(destination, CultureInfo.CurrentCulture));
+                Assert.Equal(expected, destination.ToArray());
+                Assert.Equal(expected, source.ToArray());
+            }
+            {                
+                ReadOnlySpan<char> source = a;
+                Span<char> destination = a;
+                Assert.Equal(source.Length, source.ToLowerInvariant(destination));
+                Assert.Equal(expected, destination.ToArray());
+                Assert.Equal(expected, source.ToArray());
+            }
+        }
+
+        [Fact]
+        public static void ToLowerOverlapping()
+        {
+            var expectedSource = new char[3] { 'B', 'c', 'b' };
+            var expectedDestination = new char[3] { 'b', 'c', 'b' };
+
+            {                
+                char[] a = { 'a', 'B', 'c', 'B', 'c', 'B' };
+
+                string s1 = new string(a, 1, 3);                
+                Assert.Equal(expectedDestination, s1.ToLower(CultureInfo.CurrentCulture).ToArray());
```

Comment:
where we use Span<char> as a destination buffer for string check are bit different.